Add Edgescan connector task

edblaz opened the pull request:
Added some code for the Edgescan connector task. Feel free to reach out to me here or via email (edvardas.b@edgescan.com) any time if there's anything I could help with.

lindambrown commented on tasks/connectors/edgescan/lib/edgescan_definition.rb:
```diff
@@ -0,0 +1,37 @@
+
+        private
+
+        def cves
```
It's better to set a precedence for the identifiers and send in only the one you want used. So if cve send cve, only send cwe if no cve. Make sense?

edblaz replied:
Makes perfect sense. I made that change in this commit: https://github.com/KennaSecurity/toolkit/pull/170/commits/4da1aab47d8a0dd8a7b7d62516556fde143b27b7

Closing this for now but feel free to re-open if you think I should have gone another route

lindambrown commented on tasks/connectors/edgescan/lib/kenna_api.rb:
```diff
@@ -0,0 +1,112 @@
+        end
+
+        # Fetches existing assets tagged with these Edgescan IDs from the Kenna API
+        def fetch_assets_with_edgescan_ids(edgescan_ids)
```
let's talk about why this is needed. most of the toolkit connectors don't pull assets from Kenna. Kenna will automatically match up assets in an inbound KDI file to existing assets in Kenna.

edblaz commented:
Hey, 

I just pushed a few more changes:
- The task is no longer explicitly marking assets for removal and instead letting Kenna handle that automatically 
- Added precedence when sending CVE and CWE identifiers
- After learning about automatic inactive asset deletion, removed the ability to sync _only updates_ from a certain timestamp on
- Added the ability to provide what Edgescan API hostname you want used in case you guys ever need to hit `demo.edgescan.com` 

Let me know if you have any gripes or questions!

lindambrown commented on tasks/connectors/edgescan/lib/edgescan_vulnerability.rb:
```diff
@@ -0,0 +1,64 @@
+            "last_seen_at" => data["updated_at"],
+            "scanner_score" => data["threat"],
+            "score_map" => { "5" => "10", "4" => "8", "3" => "6", "2" => "4", "1" => "2" },
+            "status" => data["status"],
```
sorry I missed this the first time but it looks like all the scanner scores I can see are under 5. score_map isn't a field in the KDI so it is likely just getting ignored. Can you use the map to update the value you have in scanner_score?

lindambrown replied:
@edblaz

edblaz replied:
It's great that you caught that! 

The `threat` on Edgescan will always be an integer between 1 and 5. That means that what we're supplying to Kenna will always be between 2 and 10. I hope that works!
